cmake compile always stop at `Detecting C compile features - done`

### Description
cmake compile always stop at `Detecting C compile features - done`


### Your System Info
ubuntu 18.04 
cmake 3.23.0-rc3
gcc 11.2.0

### Screenshots/Testimonials
![image](https://user-images.githubusercontent.com/71162630/158350681-e1e35670-2bb8-4fab-99aa-a1d0384b8607.png)

thanks for you response. I have set proxy for my network, unfortunately it do not work. when I try `cmake ..`, error occurs as blow:
```txt
-- Using the single-header code from /home/dongyang/casbin-cpp/build/_deps/json-src/single_include/
[ 11%] Performing download step (download, verify and extract) for 'benchmark-populate'
-- File already exists but no hash specified (use URL_HASH):
  file='/home/dongyang/casbin-cpp/build/_deps/benchmark-subbuild/benchmark-populate-prefix/src/v1.5.5.zip'
Old file will be removed and new file downloaded from URL.
-- Downloading...
   dst='/home/dongyang/casbin-cpp/build/_deps/benchmark-subbuild/benchmark-populate-prefix/src/v1.5.5.zip'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/google/benchmark/archive/refs/tags/v1.5.5.zip'
CMake Error at benchmark-subbuild/benchmark-populate-prefix/src/benchmark-populate-stamp/download-benchmark-populate.cmake:170 (message):
  Each download failed!

    error: downloading 'https://github.com/google/benchmark/archive/refs/tags/v1.5.5.zip' failed
          status_code: 35
          status_string: "SSL connect error"
          log:
          --- LOG BEGIN ---
          Uses proxy env variable https_proxy == 'https://127.0.0.1:7890'
    Trying 127.0.0.1:7890...

  Connected to 127.0.0.1 (127.0.0.1) port 7890 (#0)

  ALPN, offering http/1.1

  [5 bytes data]

  TLSv1.3 (OUT), TLS handshake, Client hello (1):

  [512 bytes data]

  OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 127.0.0.1:7890

  Closing connection 0

  

          --- LOG END ---
          
    


CMakeFiles/benchmark-populate.dir/build.make:99: recipe for target 'benchmark-populate-prefix/src/benchmark-populate-stamp/benchmark-populate-download' failed
make[2]: *** [benchmark-populate-prefix/src/benchmark-populate-stamp/benchmark-populate-download] Error 1
CMakeFiles/Makefile2:82: recipe for target 'CMakeFiles/benchmark-populate.dir/all' failed
make[1]: *** [CMakeFiles/benchmark-populate.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2

CMake Error at /home/dongyang/env_of_flash/cmake-3.23.0-rc3/Modules/FetchContent.cmake:1088 (message):
  Build step for benchmark failed: 2
Call Stack (most recent call first):
  /home/dongyang/env_of_flash/cmake-3.23.0-rc3/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  /home/dongyang/env_of_flash/cmake-3.23.0-rc3/Modules/FetchContent.cmake:1217 (cmake_language)
  /home/dongyang/env_of_flash/cmake-3.23.0-rc3/Modules/FetchContent.cmake:1260 (FetchContent_Populate)
  cmake/modules/Findbenchmark.cmake:23 (FetchContent_MakeAvailable)
  cmake/modules/FindExtPackages.cmake:37 (find_package)
  CMakeLists.txt:88 (include)


-- Configuring incomplete, errors occurred!
See also "/home/dongyang/casbin-cpp/build/CMakeFiles/CMakeOutput.log".
See also "/home/dongyang/casbin-cpp/build/CMakeFiles/CMakeError.log".
```

--- a/cmake/modules/Findjson.cmake
+++ b/cmake/modules/Findjson.cmake
@@ -14,16 +14,12 @@
 #  limitations under the License.
 include(FetchContent)
 
-FetchContent_Declare(json
-  GIT_REPOSITORY https://github.com/nlohmann/json.git
-  GIT_TAG v3.10.1)
-
-set(JSON_BuildTests OFF CACHE INTERNAL "")
 set(JSON_Install ON)
-FetchContent_GetProperties(json)
-FetchContent_MakeAvailable(json)
 
-if(NOT json_POPULATED)
-  FetchContent_Populate(json)
-  add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+FetchContent_Declare(json
+  GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent
+  GIT_PROGRESS TRUE
+  GIT_SHALLOW TRUE
+  GIT_TAG v3.10.5)
+
+FetchContent_MakeAvailable(json)

